### PR TITLE
Custom jvt cuts need to use detector-eta

### LINKED ISSUE #1249: Bug in JVT cut in JetSelector
Hello,

I just want to report that the JVT cut should be updated to use DetectorEta instead of eta at EM/Consit scale, since that eta is no longer the detector eta in R21:
https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/JetSelector.cxx#L1109

I never used that option but there might be someone using it.

Best,
Jona

### PR DESCRIPTION
**Pull request by kratsg**
This resolves #1249. Constituents are origin-corrected by default in R21 which means that the eta of a jet at the EM/Constituent scale is the origin-corrected eta. Custom Jvt cuts (and fJvt cuts) need to be applied using DetectorEta which is an available decoration on jets. This is a migration change from R20 where origin-correction was not applied at the constituent level, so that `jet->eta()` returned the DetectorEta if the jet was at the EM-scale.

### CODE REVIEW

File: Root/JetSelector.cxx [hunk 1]
```diff
@@ -1096,32 +1096,26 @@ int JetSelector :: PassCuts( const xAOD::Jet* jet ) {
...
-            ANA_MSG_DEBUG(" upper JVTCut is " << m_JVTCut << " - cutting this jet!!" );
-            return 0;
-          } else {
-            ANA_MSG_DEBUG(" upper JVTCut is " << m_JVTCut << " - jet passes JVT ");
-          }
-        }
-      }
+    // NB: origin-correction is applied at constituent level. Eta for Jvt needs to be the DetectorEta explicitly.
+    float jet_pt = jet->pt();
+    float jet_eta = jet->getAttribute<float>("DetectorEta");
```
Comment (mattleblanc) [inline Root/JetSelector.cxx]: Should we check to make sure it's there? Sometimes this doesn't make it into the DAODs ... maybe have everything crash & burn if it's missing.

Comment (kratsg) [reply Root/JetSelector.cxx]: If the Jvt crashes, so will this. jvt doesn't protect against it does it?

Comment (kratsg) [reply Root/JetSelector.cxx]: I'll leave it as it is. It'll crash if someone's not using it right.

**Comment by kalderon** [comment]
I noticed that this merge request inverts the behaviour of any custom JVT cuts given, ie requires JVT *less* than the value given if m_JVTcut is greater than 0. This is somewhat problematic problematic...

For example, having a JetSelector with:
```
"m_cleanJets"               :   True,
"m_doJVT"                   :   True,
"m_JVTCut"                  :   0.59,
```
which worked fine before, using the default JVT cut for the medium working point, I now get:

> JetSelector.JetSelecto...DEBUG   In pass cuts
> JetSelector.JetSelecto...DEBUG   Checking Jvt cut for jet pT=307842 MeV, DetectorEta=0.294441, and Jvt=1
> JetSelector.JetSelecto...DEBUG   Custom JVT working point with pT<60000, |eta|<2.4, Jvt<0.59
> JetSelector.JetSelecto...DEBUG    ... jet does not pass Jvt cut

which is definitely counter to my intentions...

If I remove the "m_JVTCut" and pick up the default Medium working point without trying to specify exactly that, then it works as expected. If I set the cut value to -0.59, just out of curiosity, then again it has behaviour that makes sense (JVT<-0.59 fails, JVT>-0.59 passes). 

This seems a much bigger change than just swapping detector eta for jet eta. Was this intentional?

**Pull request merged**